### ISSUE TITLE
Make Exceptions Serializable.

### ISSUE THREAD

**Issue by nathan-gs** [Azure/azure-event-hubs-java#222]
## Actual Behavior
1. 
```
User class threw exception: org.apache.spark.SparkException: Job aborted due to stage failure: Task 7.0 in stage 3.0 (TID 248) had a not serializable result: com.microsoft.azure.eventhubs.SenderContext 
Serialization stack: 
- object not serializable (class: com.microsoft.azure.eventhubs.SenderContext, value: NS: emsareingest.servicebus.windows.net, PATH: reingest-spark, REFERENCE_ID: d41016e323b64e6cbec09cdea74a1092_G9, LINK_CREDIT: 286) 
- writeObject data (class: java.lang.Throwable) 
- object (class com.microsoft.azure.eventhubs.ServerBusyException, com.microsoft.azure.eventhubs.ServerBusyException: The request was terminated because the entity is being throttled. Error code : 50002. Please wait 4 seconds and try again. Reference:eb81b973-c517-45fa-9f28-6baa1f5180d3, TrackingId:1118ae140000b9a000a694215a65d133_G9_B28, SystemTracker:emsareingest:eventhub:reingest-spark~19116, Timestamp:1/22/2018 12:05:30 PM, errorContext[NS: emsareingest.servicebus.windows.net, PATH: reingest-spark, REFERENCE_ID: d41016e323b64e6cbec09cdea74a1092_G9, LINK_CREDIT: 286]) 
- field (class: scala.util.Failure, name: exception, type: class java.lang.Throwable) 
- object (class scala.util.Failure, Failure(com.microsoft.azure.eventhubs.ServerBusyException: The request was terminated because the entity is being throttled. Error code : 50002. Please wait 4 seconds and try again. Reference:eb81b973-c517-45fa-9f28-6baa1f5180d3, TrackingId:1118ae140000b9a000a694215a65d133_G9_B28, SystemTracker:emsareingest:eventhub:reingest-spark~19116, Timestamp:1/22/2018 12:05:30 PM, errorContext[NS: emsareingest.servicebus.windows.net, PATH: reingest-spark, REFERENCE_ID: d41016e323b64e6cbec09cdea74a1092_G9, LINK_CREDIT: 286])) 
```

The issue is not about the `com.microsoft.azure.eventhubs.ServerBusyException` exception itself, but about an Exception containing properties that are not serializable.

## Expected Behavior
1. Serializable error messages 

## Versions  
- OS platform and version: Java 8u151, Spark 2.1.1, Scala 2.11.8
- Maven package version or commit ID: 0.15.1

**Comment by sjkwak** [comment]
@nathan-gs thanks for reporting the issue. We will fix the issue in coming release.

**Comment by SreeramGarlapati** [comment]
addressed in #244
